File: relationship-manager/notification-service/src/main/java/com/bank/rm/notification/controller/NotificationController.java
package com.bank.rm.notification.controller;

import com.bank.rm.common.dto.ApiResponse;
import com.bank.rm.notification.dto.NotificationDtos.*;
import com.bank.rm.notification.service.NotificationService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/notifications")
public class NotificationController {
    private final NotificationService notificationService;

    public NotificationController(NotificationService notificationService) {
        this.notificationService = notificationService;
    }

    @PostMapping
    public ResponseEntity<ApiResponse<NotificationResponse>> send(@RequestBody SendNotificationRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(ApiResponse.ok(notificationService.send(request)));
    }
}

File: relationship-manager/notification-service/src/main/kotlin/com/bank/rm/notification/controller/NotificationController.kt
package com.bank.rm.notification.controller

import com.bank.rm.common.dto.ApiResponse
import com.bank.rm.notification.dto.*
import com.bank.rm.notification.service.NotificationRouter
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/notifications")
class NotificationController(private val notificationRouter: NotificationRouter) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun send(@Valid @RequestBody request: SendNotificationRequest): ApiResponse<NotificationResponse> =
        ApiResponse(success = true, data = notificationRouter.send(request))

    @GetMapping("/{customerId}")
    fun getHistory(@PathVariable customerId: UUID): ApiResponse<List<NotificationResponse>> =
        ApiResponse(success = true, data = notificationRouter.getNotificationHistory(customerId))
}
